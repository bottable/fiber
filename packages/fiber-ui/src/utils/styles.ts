import {
  compose,
  space,
  color,
  typography,
  height,
  SpaceProps,
  ColorProps,
  TypographyProps
} from 'styled-system'

export const styleComposition = compose(space, height, color, typography)

export type StyleProps = SpaceProps & ColorProps & TypographyProps
