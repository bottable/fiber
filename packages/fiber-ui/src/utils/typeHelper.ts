export function withProperties<A, B>(component: A, properties: B): A & B {
  Object.keys(properties).forEach((key) => {
    // @ts-ignore
    component[key] = properties[key]
  })
  return component as A & B
}
