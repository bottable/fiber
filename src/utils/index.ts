export * from './typeHelper'
export * from './mergeProps'
