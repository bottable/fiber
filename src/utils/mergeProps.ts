import React from 'react'

export type MergeElementProps<
  T extends React.ElementType,
  P extends object = {}
> = Omit<React.ComponentPropsWithRef<T>, keyof P> & P
