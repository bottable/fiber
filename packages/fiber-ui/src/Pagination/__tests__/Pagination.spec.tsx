import { Pagination } from '..'

import React from 'react'
import { render } from 'test/utils'

describe('Pagination', () => {
  it('renders correctly', () => {
    const pagination = render(<Pagination />)

    expect(pagination).toMatchSnapshot()
  })
})
