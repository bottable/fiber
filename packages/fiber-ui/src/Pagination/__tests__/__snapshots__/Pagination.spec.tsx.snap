// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
<ul
  className="sc-bYwzba fuFICg"
>
  <li
    className="sc-kLokih gNeNUQ"
  >
    <button
      className="sc-pNWxx sc-dIsAE bDvTMY sc-iklIKw cqYcHR"
      disabled={true}
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={false}
    >
      <span
        className="sc-jrsJCI cjKivY"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
          />
        </svg>
      </span>
    </button>
  </li>
  <li
    className="sc-kLokih gNeNUQ"
  >
    <button
      className="sc-pNWxx sc-dIsAE imZQya sc-iklIKw cqYcHR"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={false}
    >
      <span
        className="sc-jrsJCI cjKivY"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </span>
    </button>
  </li>
</ul>
`;
